### PR TITLE
close #5 add Signin-Signup pages

### LINKED ISSUE #5: Add: SignIn/SignUp Modules
Is your Feature Request Related to a Problem? Please Describe.
N/A

------

Describe how the solution should behave
The modules should be inside the components directory and called inside the navbar because they will be accessible from there.

------

Screenshots of the design/if any

-------

Additional context
Use formik to handle validation.

### PR DESCRIPTION
**Pull request by orkunseyman**
- I created sign-in and sign-up pages with using formik for validation.
- These pages are responsive.
- Also I used yup package for  validation schema. 
Screenshot_20220408_172309
Screenshot_20220408_172346
Screenshot_20220408_172321
Screenshot_20220408_172329

### CODE REVIEW

File: src/pages/SignIn.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  AiFillLinkedin,
+  AiFillGithub,
+} from 'react-icons/ai';
+import { IoLogoDiscord } from 'react-icons/io5';
+import SignInForm from '../components/SignIn/SignInForm';
+
+function SignIn() {
+  return (
+    <div className="flex sm:grid h-100 mb-16">
+      <div className="flex justify-center w-full sm:grid grid-cols-7 gap-2">
```
Comment (haticecs) [inline src/pages/SignIn.jsx]: The left part of the pages are common in the two pages. You maybe put them in a component and use them inside the `SignIn `and `SignUp` pages

**Review by haticecs — COMMENTED** [review]
I am in love with this page. I could see one change to make. You did very good job here.

File: src/App.jsx [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import './App.css';
+import SignIn from './pages/SignIn';
+import SignUp from './pages/SignUp';
```
Comment (Iseluin) [inline src/App.jsx]: Those pages should live in a file structure like this for example: pages/SignInPage/<filename>. Please alter that.

File: src/index.css [hunk 3]
```diff
@@ -1,3 +1,8 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+@layer utilities {
+  .rectangle {
+    clip-path: polygon(0 0, 100% 0, 82% 100%, 0% 100%);
+  }
+}
```
Comment (Iseluin) [inline src/index.css:8]: Does this makes the polygon we see on the "Dreams Will Come True" part ?

File: src/pages/SignIn.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import {
+  AiFillTwitterCircle,
+  AiFillLinkedin,
+  AiFillGithub,
+} from 'react-icons/ai';
+import { IoLogoDiscord } from 'react-icons/io5';
+import SignInForm from '../components/SignIn/SignInForm';
```
Comment (Iseluin) [inline src/pages/SignIn.jsx]: Please leave a blank line between global and local imports.

File: src/pages/SignIn.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            </a>
+            <a
+              href="https://github.com/"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <AiFillGithub className="text-[#6A2C70] text-7xl hover:text-[#803886]" />
+            </a>
+          </div>
+        </div>
```
Comment (Iseluin) [inline src/pages/SignIn.jsx]: As @haticecs  said this section is repeating itself on the SignUp component, so to write DRY code, let us make this another component and use it in both modules.

File: src/pages/SignUp.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            </a>
+            <a
+              href="https://github.com/"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <AiFillGithub className="text-[#6A2C70] text-7xl hover:text-[#803886]" />
+            </a>
+          </div>
+        </div>
```
Comment (Iseluin) [inline src/pages/SignUp.jsx]: Same here.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Amazing work @orkunseyman, these modules look gorgeous. I left some little notes down below please alter your code accordingly. Aside that,

1- You can delete the CounterAPI since it gives errors when we run the page.
2- Please add tests and it looks like we will go ahead and merge this.

File: src/index.css (hunk 3 above)
Comment (orkunseyman) [reply src/index.css:8]: Yes. This is for shaping the left div.

File: src/App.jsx [hunk 7]
```diff
@@ -1,9 +1,12 @@
 import './App.css';
+import SignIn from './pages/SignInPage';
+import SignUp from './pages/SignUpPage';
 
 function App() {
   return (
     <div>
-      <h1 className="text-3xl font-bold underline">Hello world!</h1>
+      <SignIn />
+      <SignUp />
```
Comment (Iseluin) [inline src/App.jsx]: Can we put those pages inside their own folders also? src/pages/SignInPage/<your-filename>

File: src/components/LeftSignInUp/LeftSignInUp.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import {
+  AiFillTwitterCircle,
+  AiFillLinkedin,
+  AiFillGithub,
+} from 'react-icons/ai';
+import { IoLogoDiscord } from 'react-icons/io5';
+
+function LeftSignInUp() {
```
Comment (Iseluin) [inline src/components/LeftSignInUp/LeftSignInUp.jsx]: This is a confusing name for a component 🗡️  Can we change it to something more meaningful?

**Review by Iseluin — CHANGES_REQUESTED** [review]
Good work Orkun. Can you please check my comments resolved after you alter your code according to them ?

File: src/components/LeftSignInUp/LeftSignInUp.jsx (hunk 8 above)
Comment (orkunseyman) [reply src/components/LeftSignInUp/LeftSignInUp.jsx]: I thought about it but nothing came to my mind.🤔 Do you have any suggestion?

Comment (bayogluteoman) [reply src/components/LeftSignInUp/LeftSignInUp.jsx]: What about WelcomeContent ?

Comment (Iseluin) [reply src/components/LeftSignInUp/LeftSignInUp.jsx]: AuthenticationModuleSpan might be something to go with.

File: src/App.jsx [hunk 9]
```diff
@@ -1,9 +1,12 @@
 import './App.css';
+import SignInPage from './pages/SignInPage/SignInPage';
+import SignUpPage from './pages/SignUpPage/SignUpPage';
 
 function App() {
   return (
     <div>
-      <h1 className="text-3xl font-bold underline">Hello world!</h1>
+      <SignInPage />
+      <SignUpPage />
```
Comment (Iseluin) [inline src/App.jsx]: I know that you rendered the components to showcase your work for review but ideally these pages needs to be linked to the navbar and the button that we will add to the primary hero of the homepage. Please create an issue and assign yourself to it so you can work on the routing for these pages when the other needed components are merged to the dev branch.

File: src/components/LeftSignInUp/LeftSignInUp.jsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        </a>
+        <a href="https://github.com/" target="_blank" rel="noopener noreferrer">
+          <AiFillGithub className="text-[#6A2C70] text-7xl hover:text-[#803886]" />
+        </a>
+      </div>
+    </div>
+  );
+}
+
+export default LeftSignInUp;
```
Comment (Iseluin) [inline src/components/LeftSignInUp/LeftSignInUp.jsx]: I noticed in my second review that the responsivity of this component is a bit problematic. Please see the screenshots down below.

When vh is > 1089;
vh1089

The icon and some of the paragraph gets obstructed and it only changes to a new responsive structure when the icon gets obstructed fully. This is problematic. Same thing happens when viewport width is between 830px and 961px
vw 890

Please fix those responsivity issues.

File: src/index.js [hunk 11]
```diff
@@ -1,16 +1,12 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
-import { Provider } from 'react-redux';
 import App from './App';
-import store from './app/store';
 import * as serviceWorker from './serviceWorker';
```
Comment (Iseluin) [inline src/index.js]: Please restructure your imports like global imports at the top, a blank line and local imports at the bottom.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Hey @orkunseyman , I just checked your code again and saw some other minor issues with it.

1. There are CI errors still and we wont be able to merge it with them. Please fix them.
2. There are some errors on the dev console, please fix them also.
3. After everything, we need tests and we will probably be good to merge this.

**Comment by OzerOzturk** [comment]
It looks pretty good @orkunseyman, well done.

**Pull request merged**